libraries/standard/mqtt/src/iot_mqtt_api.c

```diff
@@ -439,19 +440,21 @@ static IotNetworkError_t _initializeNetworkConnection( const IotMqttNetworkInfo_
      * network connection. */
     if( pNetworkInfo->createNetworkConnection == true )
     {
-        networkStatus = pNetworkInfo->pNetworkInterface->create( pNetworkInfo->u.setup.pNetworkServerInfo,
-                                                                 pNetworkInfo->u.setup.pNetworkCredentialInfo,
-                                                                 pNetworkConnection );
+        status = pNetworkInfo->pNetworkInterface->create( pNetworkInfo->u.setup.pNetworkServerInfo,
+                                                          pNetworkInfo->u.setup.pNetworkCredentialInfo,
+                                                          pNetworkConnection );
 
-        if( networkStatus == IOT_NETWORK_SUCCESS )
+        if( status == IOT_NETWORK_SUCCESS )
         {
             /* This MQTT connection owns the network connection it created and
              * should destroy it on cleanup. */
             *pCreatedNetworkConnection = true;
         }
         else
         {
-            IOT_SET_AND_GOTO_CLEANUP( networkStatus );
+            IotLogError( "Failed to create network connection: %d", status );
+
+            IOT_SET_AND_GOTO_CLEANUP( status );
```

Comment:
Just `IOT_GOTO_CLEANUP();` since `status` is what gets set.

Reply:
Changed